frontend/src/component/Sidebar/WorkspaceListItem.jsx

```diff
@@ -63,10 +81,36 @@ class WorkspaceListItem extends React.Component {
 
   activeDropdownMenu = () => this.setState({ dropdownMenuIsActive: true })
 
+  handleClickSpace = () => {
+    this.handleReadSpaceNotifications()
+
+    if (isMobile) {
+      this.props.onClickToggleSidebar()
+    }
+  }
+
   handleMouseEnterItem = () => this.setState({ showDropdownMenuButton: true })
 
   handleMouseLeaveItem = () => this.setState(prev => ({ showDropdownMenuButton: isMobile ? true : prev.dropdownMenuIsActive }))
 
+  handleReadSpaceNotifications = async () => {
+    const { props } = this
+
+    const notificationToRead = getNotificationList(props.notificationPage).filter(n => !n.mention && !n.read).map(n => n.id)
+
+    await Promise.all(notificationToRead.map(async (notificationId) => {
```

Comment:
Why did you use `Promise.all` here?

Reply:
I have to go through every notificationId in asynchrone way so I can put the notification as read. However we can't do a foreach asyncrhone. 
There is two ways to achieve an asynchrone loop:
 - The ``for(const X of Y)`` and do your await into the for -> This will allow you to achieve what you want sequencially
 - The ``await Promise.all`` -> This will allow you to achieve what you want in parallel

[Check this answer on stackoverflow](https://stackoverflow.com/a/37576787)

Reply:
thanks!